### ISSUE TITLE
Serialization uses GPU even with CPU flag

### ISSUE THREAD

**Issue by Thrandis** [mila-udem/blocks#1152]
```
THEANO_FLAGS=device=cpu python -c 'from blocks.serialization import dump'
/Tmp/lisa/os_v5/anaconda/lib/python2.7/site-packages/skcuda/cublas.py:273: UserWarning: creating CUBLAS context to get version number warnings.warn('creating CUBLAS context to get version number')
```
Basically if I try to use dump on CPU, blocks is still initializing a free GPU. I think that
```python
try:
    import pygpu
except:
    pygpu = None
```
should be depending on the `THEANO_FLAGS`, and `pygpu` should not be imported when the flag is not set to `cuda`.

**Comment by dwf** [comment]
Agreed.

**Comment by dwf** [comment]
Fixed by #1149.
